packages/manager/src/features/CloudPulse/Utils/CloudPulseWidgetUtils.ts

```diff
@@ -456,9 +456,17 @@ export const getDimensionName = (props: DimensionNameProperties): string => {
     }
 
     if (key === 'linode_id') {
-      const linodeLabel =
-        resources.find((resource) => resource.entities?.[value] !== undefined)
-          ?.entities?.[value] ?? value;
+      let linodeLabel = value;
+      if (serviceType === 'firewall') {
+        linodeLabel =
+          resources.find((resource) => resource.entities?.[value] !== undefined)
+            ?.entities?.[value] ?? value;
+      }
+      if (serviceType === 'blockstorage') {
+        linodeLabel =
+          resources.find((resource) => resource.volumeLinodeId === value)
+            ?.volumeLinodeLabel ?? value;
```

Comment:
```suggestion
            ?.volumeLinodeLabel ?? linodeLabel;
```

anyway you assigned value to linodeLabel at top, then why duplicate assignment